generation/internal/php.go

```diff
@@ -0,0 +1,55 @@
+package internal
+
+import (
+	"github.com/CircleCI-Public/circleci-config/config"
+	"github.com/CircleCI-Public/circleci-config/labeling/labels"
+)
+
+func GeneratePHPJobs(ls labels.LabelSet) []*Job {
+	if !ls[labels.DepsPhp].Valid {
+		return nil
+	}
+
+	jobs := []*Job{}
+
+	if hasLib(ls, "phpunit/phpunit") {
+		jobs = append(jobs, phpunitJob(ls))
+	}
+	return jobs
+}
+
+func hasLib(ls labels.LabelSet, lib string) bool {
+	if ls[labels.DepsPhp].Dependencies[lib] != "" {
+		return true
+	}
+	return false
+}
+
+func initialPhpSteps(ls labels.LabelSet) []config.Step {
+	checkout := checkoutStep(ls[labels.DepsPhp])
+	installPackages := config.Step{
+		Type:    config.OrbCommand,
+		Command: "php/install-packages",
+	}
+	return []config.Step{checkout, installPackages}
+}
+
+func phpunitJob(ls labels.LabelSet) *Job {
+	steps := initialPhpSteps(ls)
+	steps = append(steps, config.Step{
+		Type:    config.Run,
+		Name:    "run tests",
+		Command: "./vendor/bin/phpunit",
+	})
+	return &Job{
+		Job: config.Job{
+			Name:         "test-php",
+			Comment:      "Install php packages and run tests",
+			Steps:        steps,
+			DockerImages: []string{"cimg/php:8.2.7-node"},
+		},
+		Orbs: map[string]string{
+			"php": "circleci/php@1.1.0",
```

Comment:
The orb seems to assume PHP 7.x: https://github.com/CircleCI-Public/php-orb/blob/master/src/commands/install-php.yml#L6
Is there potential for a compatibility issue with the php 8.x image?

It seems like PHP version can be specified in the `composer.json` file: https://getcomposer.org/doc/04-schema.md#package-links

Reply:
I think the orb commands we are using to install packages should work even in a later image. Determining the correct php version will be a challenge, particularly for repos that don't put the version in the composer.json. 

I'm going to leave looking looking for the version in composer.json for a future revision.